main/NotificationSystem.cpp

```diff
@@ -135,14 +135,8 @@ bool CNotificationSystem::Unregister(CNotificationObserver* pNotifier)
 	if (m_notifiers.size() > 0)
 	{
 		boost::unique_lock<boost::shared_mutex> lock(m_mutex);
-		for (size_t i = 0; i < m_notifiers.size(); i++)
-		{
-			if (m_notifiers[i] == pNotifier)
-			{
-				m_notifiers.erase(m_notifiers.begin() + i);
-				return true;
-			}
-		}
+		m_notifiers.erase(std::remove_if(m_notifiers.begin(), m_notifiers.end(), [=](CNotificationObserver *notifier) { return notifier == pNotifier; }), m_notifiers.end());
```

Comment:
Not sure what is clearer to read ?
Maybe better to not write everything on a single line

Reply:
It reduces it to https://en.wikipedia.org/wiki/Erase%E2%80%93remove_idiom

which I think is easier to reason about.

Reply:
True, but for readability I would say it does not make things clearer
The for loop could be rewritten to 'auto', but the erase is a- not clear to read, b-does not return true if erased
First the possible item is erased, then you check if the item is in the array